js/src/vm/PortableBaselineInterpret.cpp

```diff
@@ -1734,7 +1772,9 @@ ICInterpretOps(BaselineFrame* frame, VMFrameManager& frameMgr, State& state,
     }
   });
   INT32_OP(BitOr, |, {});
+  INT32_OP(BitXor, ^, {});
   INT32_OP(BitAnd, &, {});
+  INT32_OP(LeftShift, <<, {});
```

Comment:
It looks like the masm implementation (`flexibleLshift32`) wraps the shift amount (`amt & 0x1f` or similar); afaik, C++ `<<` is undefined for shifts greater than width, so I think we need to do the same here?

Reply:
Ah, you're right. Their x86 implementation misled me because x86 implicitly masks the shift amount, but the arm32 implementation emits an explicit bitwise-and instruction.

Since the `INT32_OP` helper currently assumes we can directly use C++ operators, I'll drop left-shift from this PR and come back to it later. I was already skipping right-shift for similar reasons.